api4/user.go

```diff
@@ -1191,6 +1193,64 @@ func generateMfaSecret(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(secret.ToJson()))
 }
 
+func generateRecovery(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if c.App.Session.IsOAuth {
+		c.SetPermissionError(model.PERMISSION_EDIT_OTHER_USERS)
```

Comment:
I think the permission used for the error itself doesn't match, since you don't try to edit other use data here. Throwing a bad request error e.g. makes more sense here.